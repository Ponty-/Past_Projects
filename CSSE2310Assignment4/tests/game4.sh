#!/bin/bash

if [ $# != 4 ]
then
    echo "Usage: $0 binarydir port resultdir correct_dir" 
    exit 1
fi



bindir=$1
port=$2
res=$3
right=$4
name=$0

expected=0

if [ ! -f $bindir/client499 ]
then
    echo "$bindir/client499 is missing"
    exit 1
fi

if [ ! -f $bindir/serv499 ]
then
    echo "$bindir/serv499 is missing"
    exit 1
fi



$bindir/serv499 $port '%s%s%s%s%s' 4.deck &
spid=$!
$bindir/client499 zzzzzzzz gg $port > $res/$name.a.out 2>$res/$name.a.err < ax &
pid1=$!
$bindir/client499 123456 gg $port > $res/$name.b.out 2>$res/$name.b.err < bx &
pid2=$!
$bindir/client499 CCC gg $port > $res/$name.c.out 2>$res/$name.c.err < cx &
pid3=$!
$bindir/client499 dddd gg $port > $res/$name.d.out 2>$res/$name.d.err < dx &
pid4=$!

wait $pid1 $pid2 $pid3 $pid4 2> /dev/null

$bindir/client499 zzzzA game $port > $res/$name.a.2.out 2>$res/$name.a.2.err < ax &
pid1=$!
$bindir/client499 123B game $port > $res/$name.b.2.out 2>$res/$name.b.2.err < bx &
pid2=$!
$bindir/client499 C game $port > $res/$name.c.2.out 2>$res/$name.c.2.err < cx &
pid3=$!
$bindir/client499 D game $port > $res/$name.d.2.out 2>$res/$name.d.2.err < dx &
pid4=$!

wait $pid1 $pid2 $pid3 $pid4 2> /dev/null



kill $spid 2> /dev/null
wait $spid 2> /dev/null

#if [ $result != $expected ]
#then
#    echo "Expected $expected got $result"
#    exit 1
#fi

for v in a b c d
do

diff -q $res/$name.$v.2.out $right/$name.$v.2.out > /dev/null
if [ $? != 0 ]
then
   echo "Stdout mismatch for client $v"
   echo "diff $res/$name.$v.2.out $right/$name.$v.2.out"
   exit 2
fi

diff -q $res/$name.$v.2.err $right/$name.$v.2.err > /dev/null
if [ $? != 0 ]
then
   echo "Stderr mismatch for client $v"
   echo "diff $res/$name.$v.2.err $right/$name.$v.2.err"
   exit 3
fi

done

echo OK
exit 0

