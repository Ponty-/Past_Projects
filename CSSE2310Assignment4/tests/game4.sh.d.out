Info: %s%s%s%s%s
Info: Team1: 123456, dddd
Info: Team2: CCC, zzzzzzzz
S: T 6
C: A T 6 2
D: J 9 6 3
H: K 9 4
Info: 123456 bids 7H
Info: CCC passes
[7H] - Bid (or pass)> [7H] - Bid (or pass)> Info: zzzzzzzz passes
Info: 123456 passes
S: T 6
C: A T 6 2
D: J 9 6 3
H: K 9 4
Lead> Info: zzzzzzzz plays 9S
Info: 123456 plays AS
Info: CCC plays 7S
Info: 123456 won
Info: 123456 plays KS
Info: CCC plays QS
S: 6
C: A T 6 2
D: J 9 6 3
H: K 9 4
[S] play> Info: zzzzzzzz plays 4S
Info: 123456 won
Info: 123456 plays QH
Info: CCC plays AH
S:
C: A T 6 2
D: J 9 6 3
H: K 9 4
[H] play> Info: zzzzzzzz plays 5H
Info: CCC won
Info: CCC plays QC
S:
C: A T 6 2
D: J 9 6 3
H: K 9
[C] play> Info: zzzzzzzz plays 4C
Info: 123456 plays 5C
Info: CCC won
Info: CCC plays KD
S:
C: A T 6
D: J 9 6 3
H: K 9
[D] play> S:
C: A T 6
D: J 9 6 3
H: K 9
[D] play> Info: zzzzzzzz plays 2D
Info: 123456 plays JS
Info: 123456 won
Info: 123456 plays 7H
Info: CCC plays TH
S:
C: A T 6
D: J 9 6
H: K 9
[H] play> S:
C: A T 6
D: J 9 6
H: K 9
[H] play> Info: zzzzzzzz plays 8H
Info: CCC won
Info: CCC plays 7C
S:
C: A T 6
D: J 9 6
H: K
[C] play> Info: zzzzzzzz plays 9C
Info: 123456 plays 8C
Info: dddd won
S:
C: T 6
D: J 9 6
H: K
Lead> Info: zzzzzzzz plays JH
Info: 123456 plays 3H
Info: CCC plays 2H
Info: dddd won
S:
C: T 6
D: J 9 6
H:
Lead> Info: zzzzzzzz plays KC
Info: 123456 plays JC
Info: CCC plays 3C
Info: zzzzzzzz won
Info: zzzzzzzz plays AD
Info: 123456 plays 2S
Info: CCC plays 5D
S:
C: 6
D: J 9 6
H:
[D] play> Info: 123456 won
Info: 123456 plays 8S
Info: CCC plays 7D
S:
C: 6
D: 9 6
H:
[S] play> Info: zzzzzzzz plays QD
Info: 123456 won
Info: 123456 plays 3S
Info: CCC plays TD
S:
C:
D: 9 6
H:
[S] play> Info: zzzzzzzz plays 4D
Info: 123456 won
Info: 123456 plays 5S
Info: CCC plays 6H
S:
C:
D: 9
H:
[S] play> S:
C:
D: 9
H:
[S] play> Info: zzzzzzzz plays 8D
Info: 123456 won
Info: Team 1=220, Team 2=0
S: T 6
C: A 8 6 2
D: J 9 6 3
H: K 9 4
Info: 123456 bids 6S
Info: CCC bids 6H
[6H] - Bid (or pass)> Info: zzzzzzzz passes
Info: 123456 passes
Info: CCC plays AH
S: T 6
C: A 8 6 2
D: J 9 6 3
H: K 9 4
[H] play> Info: zzzzzzzz plays 5H
Info: 123456 plays 3H
Info: CCC won
Info: CCC plays KD
S: T 6
C: A 8 6 2
D: J 9 6 3
H: 9 4
[D] play> Info: zzzzzzzz plays AD
Info: 123456 plays 2S
Info: zzzzzzzz won
Info: zzzzzzzz plays KC
Info: 123456 plays 5C
Info: CCC plays 3C
S: T 6
C: A 8 6 2
D: J 9 6
H: 9 4
[C] play> Info: zzzzzzzz won
Info: zzzzzzzz plays QD
Info: 123456 plays TC
Info: CCC plays 5D
S: T 6
C: A 8 6
D: J 9 6
H: 9 4
[D] play> S: T 6
C: A 8 6
D: J 9 6
H: 9 4
[D] play> Info: zzzzzzzz won
Info: zzzzzzzz plays 9S
Info: 123456 plays AS
Info: CCC plays 7S
S: T 6
C: A 8 6
D: J 9
H: 9 4
[S] play> Info: 123456 won
Info: 123456 plays KS
Info: CCC plays QS
S: T
C: A 8 6
D: J 9
H: 9 4
[S] play> Info: zzzzzzzz plays 4S
Info: 123456 won
Info: 123456 plays 3S
Info: CCC plays TH
S:
C: A 8 6
D: J 9
H: 9 4
[S] play> Info: zzzzzzzz plays 2D
Info: CCC won
Info: CCC plays QC
S:
C: A 8
D: J 9
H: 9 4
[C] play> Info: zzzzzzzz plays 4C
Info: 123456 plays JC
Info: dddd won
S:
C: 8
D: J 9
H: 9 4
Lead> Info: zzzzzzzz plays 8H
Info: 123456 plays 7H
Info: CCC plays 2H
Info: zzzzzzzz won
Info: zzzzzzzz plays JH
Info: 123456 plays QH
Info: CCC plays 6H
S:
C: 8
D: J 9
H: 9
[H] play> Info: 123456 won
Info: 123456 plays 5S
Info: CCC plays 7D
S:
C: 8
D: J 9
H:
[S] play> Info: zzzzzzzz plays 4D
Info: 123456 won
Info: 123456 plays 8S
Info: CCC plays 9C
S:
C: 8
D: J
H:
[S] play> Info: zzzzzzzz plays 7C
Info: 123456 won
Info: 123456 plays JS
Info: CCC plays TD
S:
C:
D: J
H:
[S] play> Info: zzzzzzzz plays 8D
Info: 123456 won
Info: Team 1=220, Team 2=150
S: 9 4
C: K 9 4
D: A Q 8 4 2
H: J 8 5
Info: 123456 bids 6S
Info: CCC bids 7H
[7H] - Bid (or pass)> Info: zzzzzzzz passes
Info: 123456 passes
Info: CCC plays AC
S: 9 4
C: K 9 4
D: A Q 8 4 2
H: J 8 5
[C] play> Info: zzzzzzzz plays 5C
Info: 123456 plays 3C
Info: CCC won
Info: CCC plays TC
S: 9 4
C: 9 4
D: A Q 8 4 2
H: J 8 5
[C] play> Info: zzzzzzzz plays 8C
Info: 123456 plays 7C
Info: CCC won
Info: CCC plays KH
S: 9 4
C: 9
D: A Q 8 4 2
H: J 8 5
[H] play> Info: zzzzzzzz plays 3H
Info: 123456 plays 2H
Info: CCC won
Info: CCC plays JD
S: 9 4
C: 9
D: A Q 8 4 2
H: J 8
[D] play> Info: zzzzzzzz plays 2S
Info: 123456 plays 5D
Info: CCC won
Info: CCC plays 9D
S: 9 4
C: 9
D: A Q 8 4
H: J 8
[D] play> Info: zzzzzzzz plays 3S
Info: 123456 plays 7D
Info: CCC won
Info: CCC plays 9H
S: 9 4
C: 9
D: A Q 8
H: J 8
[H] play> S: 9 4
C: 9
D: A Q 8
H: J 8
[H] play> Info: zzzzzzzz plays 7H
Info: 123456 plays AH
Info: 123456 won
Info: 123456 plays QC
Info: CCC plays 2C
S: 9 4
C: 9
D: A Q 8
H: 8
[C] play> Info: zzzzzzzz plays JC
Info: 123456 won
Info: 123456 plays 6H
Info: CCC plays 4H
S: 9 4
C:
D: A Q 8
H: 8
[H] play> Info: zzzzzzzz plays QH
Info: zzzzzzzz won
Info: zzzzzzzz plays 5S
Info: 123456 plays QS
Info: CCC plays 6S
S: 9 4
C:
D: A Q 8
H:
[S] play> Info: 123456 won
Info: 123456 plays 7S
Info: CCC plays TS
S: 9
C:
D: A Q 8
H:
[S] play> Info: zzzzzzzz plays 8S
Info: CCC won
Info: CCC plays 6C
S:
C:
D: A Q 8
H:
[C] play> Info: zzzzzzzz plays JS
Info: 123456 plays TD
Info: CCC won
Info: CCC plays 3D
S:
C:
D: A Q
H:
[D] play> Info: zzzzzzzz plays KS
Info: 123456 plays KD
Info: dddd won
S:
C:
D: Q
H:
Lead> Info: zzzzzzzz plays AS
Info: 123456 plays TH
Info: CCC plays 6D
Info: 123456 won
Info: Team 1=220, Team 2=350
S: T 6
C: A T 6 2
D: J 9 6 3
H: K 9 4
Info: 123456 bids 9H
Info: 123456 plays 2S
Info: CCC plays QS
S: T 6
C: A T 6 2
D: J 9 6 3
H: K 9 4
[S] play> Info: zzzzzzzz plays 4S
Info: CCC won
Info: CCC plays 3C
S: 6
C: A T 6 2
D: J 9 6 3
H: K 9 4
[C] play> Info: zzzzzzzz plays KC
Info: 123456 plays 5C
Info: dddd won
S: 6
C: T 6 2
D: J 9 6 3
H: K 9 4
Lead> Info: zzzzzzzz plays 5H
Info: 123456 plays 3H
Info: CCC plays AH
Info: CCC won
Info: CCC plays 7S
S: 6
C: T 6 2
D: J 9 6 3
H: 9 4
[S] play> Info: zzzzzzzz plays 9S
Info: 123456 plays 3S
Info: zzzzzzzz won
Info: zzzzzzzz plays AD
Info: 123456 plays 8C
Info: CCC plays 5D
S:
C: T 6 2
D: J 9 6 3
H: 9 4
[D] play> S:
C: T 6 2
D: J 9 6 3
H: 9 4
[D] play> Info: zzzzzzzz won
Info: zzzzzzzz plays 9C
Info: 123456 plays JC
Info: CCC plays 7C
S:
C: T 6 2
D: 9 6 3
H: 9 4
[C] play> Info: 123456 won
Info: 123456 plays AS
Info: CCC plays KD
S:
C: 6 2
D: 9 6 3
H: 9 4
[S] play> Info: zzzzzzzz plays 8H
Info: zzzzzzzz won
Info: zzzzzzzz plays JH
Info: 123456 plays QH
Info: CCC plays 2H
S:
C: 2
D: 9 6 3
H: 9 4
[H] play> Info: 123456 won
Info: 123456 plays 7H
Info: CCC plays 6H
S:
C: 2
D: 9 6 3
H: 9
[H] play> Info: zzzzzzzz plays 2D
Info: dddd won
S:
C: 2
D: 9 6 3
H:
Lead> Info: zzzzzzzz plays 4C
Info: 123456 plays 5S
Info: CCC plays QC
Info: CCC won
Info: CCC plays 7D
S:
C:
D: 9 6 3
H:
[D] play> Info: zzzzzzzz plays QD
Info: 123456 plays 8S
Info: zzzzzzzz won
Info: zzzzzzzz plays 8D
Info: 123456 plays JS
Info: CCC plays TD
S:
C:
D: 9 6
H:
[D] play> Info: CCC won
Info: CCC plays TH
S:
C:
D: 6
H:
[H] play> Info: zzzzzzzz plays 4D
Info: 123456 plays KS
Info: CCC won
Info: Team 1=-80, Team 2=350
S: T 6
C: A 8 6 2
D: J 9 6 3
H: K 9 4
Info: 123456 bids 9H
Info: 123456 plays 3H
Info: CCC plays AH
S: T 6
C: A 8 6 2
D: J 9 6 3
H: K 9 4
[H] play> Info: zzzzzzzz plays 5H
Info: CCC won
Info: CCC plays 7S
S: T 6
C: A 8 6 2
D: J 9 6 3
H: K 9
[S] play> Info: zzzzzzzz plays 4S
Info: 123456 plays AS
Info: 123456 won
Info: 123456 plays KS
Info: CCC plays QS
S: 6
C: A 8 6 2
D: J 9 6 3
H: K 9
[S] play> Info: zzzzzzzz plays 9S
Info: 123456 won
Info: 123456 plays 5C
Info: CCC plays QC
S:
C: A 8 6 2
D: J 9 6 3
H: K 9
[C] play> Info: zzzzzzzz plays 4C
Info: CCC won
Info: CCC plays KD
S:
C: A 8 6
D: J 9 6 3
H: K 9
[D] play> Info: zzzzzzzz plays 2D
Info: 123456 plays 7H
Info: 123456 won
Info: 123456 plays QH
Info: CCC plays TH
S:
C: A 8 6
D: J 9 6
H: K 9
[H] play> Info: zzzzzzzz plays 8H
Info: 123456 won
Info: 123456 plays 2S
Info: CCC plays 2H
S:
C: A 8 6
D: J 9 6
H: K
[S] play> Info: zzzzzzzz plays 4D
Info: CCC won
Info: CCC plays 6H
S:
C: 8 6
D: J 9 6
H: K
[H] play> Info: zzzzzzzz plays JH
Info: 123456 plays 3S
Info: dddd won
S:
C: 8 6
D: J 9 6
H:
Lead> Info: zzzzzzzz plays AD
Info: 123456 plays 5S
Info: CCC plays 5D
Info: zzzzzzzz won
Info: zzzzzzzz plays 7C
Info: 123456 plays JC
Info: CCC plays 3C
S:
C: 8 6
D: 9 6
H:
[C] play> Info: 123456 won
Info: 123456 plays 8S
Info: CCC plays 9C
S:
C: 6
D: 9 6
H:
[S] play> Info: zzzzzzzz plays KC
Info: 123456 won
Info: 123456 plays JS
Info: CCC plays 7D
S:
C: 6
D: 9
H:
[S] play> Info: zzzzzzzz plays 8D
Info: 123456 won
Info: 123456 plays TC
Info: CCC plays TD
S:
C:
D: 9
H:
[C] play> Info: zzzzzzzz plays QD
Info: 123456 won
Info: Team 1=220, Team 2=350
S: 9 4
C: K 9 4
D: A Q 8 4 2
H: J 8 5
Info: 123456 bids 9H
Info: 123456 plays AH
Info: CCC plays KH
S: 9 4
C: K 9 4
D: A Q 8 4 2
H: J 8 5
[H] play> Info: zzzzzzzz plays 3H
Info: 123456 won
Info: 123456 plays KD
Info: CCC plays JD
S: 9 4
C: K 9 4
D: A Q 8 4 2
H: J 8
[D] play> Info: zzzzzzzz plays 7H
Info: zzzzzzzz won
Info: zzzzzzzz plays QH
Info: 123456 plays 2H
Info: CCC plays 4H
S: 9 4
C: K 9 4
D: A Q 8 4
H: J 8
[H] play> S: 9 4
C: K 9 4
D: A Q 8 4
H: J 8
[H] play> Info: zzzzzzzz won
Info: zzzzzzzz plays AS
Info: 123456 plays 7S
Info: CCC plays 6S
S: 9 4
C: K 9 4
D: A Q 8 4
H: J
[S] play> Info: zzzzzzzz won
Info: zzzzzzzz plays KS
Info: 123456 plays QS
Info: CCC plays TS
S: 9
C: K 9 4
D: A Q 8 4
H: J
[S] play> Info: zzzzzzzz won
Info: zzzzzzzz plays JS
Info: 123456 plays 6H
Info: CCC plays AC
S:
C: K 9 4
D: A Q 8 4
H: J
[S] play> Info: 123456 won
Info: 123456 plays QC
Info: CCC plays TC
S:
C: K 9
D: A Q 8 4
H: J
[C] play> Info: zzzzzzzz plays 5C
Info: 123456 won
Info: 123456 plays 7C
Info: CCC plays 2C
S:
C: K
D: A Q 8 4
H: J
[C] play> S:
C: K
D: A Q 8 4
H: J
[C] play> Info: zzzzzzzz plays 8C
Info: dddd won
S:
C:
D: A Q 8 4
H: J
Lead> Info: zzzzzzzz plays 8S
Info: 123456 plays TD
Info: CCC plays 9D
Info: dddd won
S:
C:
D: Q 8 4
H: J
Lead> Info: zzzzzzzz plays 2S
Info: 123456 plays TH
Info: CCC plays 9H
Info: dddd won
S:
C:
D: Q 8 4
H:
Lead> Info: zzzzzzzz plays 3S
Info: 123456 plays 5D
Info: CCC plays 3D
Info: 123456 won
Info: 123456 plays 7D
Info: CCC plays 6D
S:
C:
D: Q 8
H:
[D] play> Info: zzzzzzzz plays JC
Info: dddd won
S:
C:
D: Q
H:
Lead> Info: zzzzzzzz plays 5S
Info: 123456 plays 3C
Info: CCC plays 6C
Info: dddd won
Info: Team 1=520, Team 2=350
Info: Winner is Team 1
